### PR TITLE
feat: add datalog-based single source shortest paths

### LINKED ISSUE #7198: Datalog example with single source shortest path
Let's try to have two small Datalog examples with:

- Single source shortest distances
- Single source shortest paths

Programs to be put into the example folder with a handful tests.

Ideally both programs look similar and straightforward to read.

Once done, update https://doc.flix.dev/lattice-semantics.html

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #7198

### CODE REVIEW

**Comment by jaschdoc** [comment]
Should we use the `Path` type from https://github.com/flix/flix/pull/7191 ?

File: examples/larger-examples/datalog/single-source-shortest-paths.flix [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    }
+
+    instance LowerBound[Path[a]] {
+        // The longest list
+        pub def minValue(): Path[a] = Bot
+    }
+
+    instance PartialOrder[Path[a]] {
+        pub def lessEqual(x: Path[a], y: Path[a]): Bool = match (x, y) {
+            case (Bot, Bot)           => true
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply examples/larger-examples/datalog/single-source-shortest-paths.flix:49]: because bot is smaller than any specific path.

**Comment by magnus-madsen** [comment]
Once this is done, lets do one with paths where the edges have distances. Then you need paths to be `Path(List[a], Int32)`.

**Comment by jaschdoc** [comment]
> Once this is done, lets do one with paths where the edges have distances. Then you need paths to be `Path(List[a], Int32)`.

Should the shortest Dist be updated with this new `Path` definition or what is your point exactly? :)

**Comment by magnus-madsen** [comment]
> > Once this is done, lets do one with paths where the edges have distances. Then you need paths to be `Path(List[a], Int32)`.
> 
> Should the shortest Dist be updated with this new `Path` definition or what is your point exactly? :)

A new program that computes shortests paths, but where each path does not have length 1, but some arbitrary length stored on the path.

**Review by magnus-madsen — APPROVED** [review]

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**